django/db/backends/base/base.py

```diff
@@ -286,15 +307,35 @@ def get_autocommit(self):
         self.ensure_connection()
         return self.autocommit
 
-    def set_autocommit(self, autocommit):
+    def set_autocommit(self, autocommit, force_begin_transaction_with_broken_autocommit=False):
```

Comment:
`force_begin_transaction_with_broken_autocommit` is a really long name, but its purposes still isn't obvious to me. Could you add some info to the docstring about its purpose? If the name could be simplified/shortened, I think that would be beneficial as well.